cpp/daal/src/algorithms/objective_function/cross_entropy_loss/cross_entropy_loss_dense_default_batch_impl.i

```diff
@@ -79,23 +80,20 @@ void CrossEntropyLossKernel<algorithmFPType, method, cpu>::applyBeta(const algor
 }
 
 template <typename algorithmFPType, Method method, CpuType cpu>
-void CrossEntropyLossKernel<algorithmFPType, method, cpu>::applyBetaThreaded(const algorithmFPType * x, const algorithmFPType * beta,
-                                                                             algorithmFPType * xb, size_t nRows, size_t nClasses, size_t nCols,
-                                                                             bool bIntercept)
-{
-    DAAL_ITTNOTIFY_SCOPED_TASK(applyBeta);
-    applyBetaImpl<algorithmFPType, cpu>(x, beta, xb, nRows, nClasses, nCols, bIntercept, true);
-}
-
-template <typename algorithmFPType, Method method, CpuType cpu>
-void CrossEntropyLossKernel<algorithmFPType, method, cpu>::softmax(const algorithmFPType * arg, algorithmFPType * res, size_t nRows, size_t nCols)
+void CrossEntropyLossKernel<algorithmFPType, method, cpu>::softmax(const algorithmFPType * const arg, algorithmFPType * const res, size_t nRows,
+                                                                   size_t nCols, algorithmFPType * const softmaxSums,
+                                                                   const algorithmFPType * const yLocal)
 {
     const algorithmFPType expThreshold = daal::internal::Math<algorithmFPType, cpu>::vExpThreshold();
+    if (softmaxSums != nullptr)
+    {
+        services::internal::service_memset<algorithmFPType, cpu>(softmaxSums, 0.0, nCols);
```

Comment:
`service_memset` - function with threading. it is necessary? Use `service_memset_seq`